### PR TITLE
Serve all source sets

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR extends Gradle plugins to support not only `main` and `test` source sets, but all in a project.

Other changes:
  * The code migrated to latest `tool-base` API.
  * Types related to annotations were pushed to the package level so that it's easy to see the architecture of the module.
  * `RejectionGenPlugin` plugin was refactored extracting a method object for creating tasks.

### CODE REVIEW

File: mc-java-rejection/src/test/java/io/spine/tools/mc/java/rejection/gradle/RejectionGenPluginTest.java [hunk 1]
```diff
@@ -26,110 +26,111 @@
...
-        String projectAbsolutePath = testProjectDir.getAbsolutePath();
-        File generatedFile = new File(projectAbsolutePath + rejectionsJavadocThrowableSource());
-        JavaClassSource generatedSource = Roaster.parse(JavaClassSource.class, generatedFile);
-        assertRejectionJavadoc(generatedSource);
-        assertBuilderJavadoc(
-                (JavaClassSource) generatedSource.getNestedType(SimpleClassName.ofBuilder().value())
-        );
+    @Nested
+    @DisplayName("place generated code under the `spine` directory for")
+    class GeneratedRoot {
```
[the remark this answers is not in the document]
Comment (alexander-yevsyukov) [reply mc-java-rejection/src/test/java/io/spine/tools/mc/java/rejection/gradle/RejectionGenPluginTest.java:66]: ignore

File: mc-java-rejection/src/test/java/io/spine/tools/mc/java/rejection/gradle/RejectionGenPluginTest.java [hunk 2]
```diff
@@ -26,110 +26,111 @@
...
     }
 
-    private static void assertMethodDoc(String expectedComment,
-                                        JavaClassSource source,
-                                        String methodName) {
-        MethodSource<JavaClassSource> method = findMethod(source, methodName);
-        assertDoc(expectedComment, method);
+    @Nested
+    @DisplayName("use the package specified in proto file options")
+    class PackageDir {
```
[the remark this answers is not in the document]
Comment (alexander-yevsyukov) [reply mc-java-rejection/src/test/java/io/spine/tools/mc/java/rejection/gradle/RejectionGenPluginTest.java:97]: ignore

File: mc-java-rejection/src/main/java/io/spine/tools/mc/java/rejection/gradle/RejectionGenAction.java [hunk 3]
```diff
@@ -27,93 +27,148 @@
...
     }
 
+    /**
+     * Obtains the predicate which accepts sources files belonging to currently served
+     * {@linkplain #ssn source set}.
+     */
+    private Predicate<SourceFile> belongsToSourceSet() {
+        @Nullable FileCollection fileCollection = Projects.protoFiles(project(), ssn);
+        checkState(fileCollection != null, "No proto files found in the source set `%s`.", ssn);
+        Set<Path> protoFiles = fileCollection.getFiles()
```
[the remark this answers is not in the document]
Comment (alexander-yevsyukov) [reply mc-java-rejection/src/main/java/io/spine/tools/mc/java/rejection/gradle/RejectionGenAction.java:129]: ignore

File: mc-java-rejection/src/test/java/io/spine/tools/mc/java/rejection/gradle/RejectionJavadocTest.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        project.executeTask(generateRejections(SourceSetName.main));
+        File generatedFile = new File(
+                projectDir.getAbsolutePath() + rejectionsJavadocThrowableSource()
+        );
+        generatedSource = Roaster.parse(JavaClassSource.class, generatedFile);
+    }
+
+    @Nested
+    @DisplayName("generate Javadoc for")
+    class GenerateJavadoc {
```
[the remark this answers is not in the document]
Comment (alexander-yevsyukov) [reply mc-java-rejection/src/test/java/io/spine/tools/mc/java/rejection/gradle/RejectionJavadocTest.java:83]: ignore

File: mc-java-annotation/src/main/java/io/spine/tools/mc/java/annotation/mark/Job.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+package io.spine.tools.mc.java.annotation.mark;
+
+import io.spine.logging.Logging;
+
+/**
+ * A source code annotation job obtains required {@link Annotator} instances from
+ * {@link AnnotatorFactory} passed to the {@link #execute(AnnotatorFactory)} method.
+ */
+public interface Job extends Logging {
```
Comment (dmdashenkov) [inline mc-java-annotation/src/main/java/io/spine/tools/mc/java/annotation/mark/Job.java:35]: Is there a need to always have `Logging` in a `Job`? Maybe, we could implement it more locally in order to make the interface more understandable.

File: mc-java-rejection/src/main/java/io/spine/tools/mc/java/rejection/gradle/ProtoModule.java [hunk 6]
```diff
@@ -59,68 +61,38 @@
...
-        FileCollection files = project.fileTree(targetDir);
-        return files;
-    }
+//    private @Nullable FileCollection protoFiles(SourceSetName ssn) {
+//        SourceSet sourceSet = sourceSet(project, ssn);
+//        @Nullable Object extension =
+//                sourceSet.getExtensions()
+//                         .findByName(sourceSetExtensionName);
+//        return (FileCollection) extension;
+//    }
```
Comment (dmdashenkov) [inline mc-java-rejection/src/main/java/io/spine/tools/mc/java/rejection/gradle/ProtoModule.java:84]: Please delete the commented code.

File: mc-java-rejection/src/test/resources/rejections-gen-plugin-test/src/test/proto/test_rejections.proto [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+syntax = "proto3";
+
+package spine.sample.rejections;
+
+option java_package = "io.spine.sample.rejections";
+option java_multiple_files = false;
+// Do not specify `java_outer_classname` option as we're not generating multiple
```
Comment (dmdashenkov) [inline mc-java-rejection/src/test/resources/rejections-gen-plugin-test/src/test/proto/test_rejections.proto:32]:
```suggestion
option java_multiple_files = false;

// Do not specify `java_outer_classname` option as we're not generating multiple
```

**Review by dmdashenkov — APPROVED** [review]

**Review by armiol — COMMENTED** [review]
@alexander-yevsyukov LGTM in general. Please see my comments.

**Pull request merged**